sql/src/main/java/io/crate/operation/collect/collectors/CompositeCollector.java

```diff
@@ -80,6 +80,7 @@
     private final List<CrateCollector> collectors;
     private final Receiver receiver;
     private final RowReceiver downstream;
+    private volatile boolean isStopped = false;
```

Comment:
why does this have to be volatile? CompositeCollector runs single threaded.

And the other state varialbes used in finish are non-sychronized/volatile as well